lib/x86/setup.c

```diff
@@ -3,6 +3,7 @@
  *
  * Copyright (C) 2017, Red Hat Inc, Andrew Jones <drjones@redhat.com>
  * Copyright (C) 2021, UC San Diego, Zixuan Wang <zxwang42@gmail.com>
+ * Copyright (C) 2021, Google Inc, Zixuan Wang <zixuanwang@google.com>
```

Comment:
This file didn't change much. Should we add the extra copyright? Rant below. But I guess you can ignore my rant :-). Regardless, I'd personally suggest to just leave the Google copyright here. Rationale being that this file really didn't change in a signifcant way from what was done during the internship. However, it's up to you. If you disagree, please leave the new Copyright in place.

Aside: I really dislike authorships and copyrights at the top of files like this. I feel the commit history reflects everyone's contributions much more accurately. As an example, look at the vmx.c file the kernel [1]:

The original authors of the files have not contributed to the file for many years. And the file has evolved to something very different from when they first authored the file. It seems really disingenuous to me that they're still listed as the authors of the file.

[1] https://elixir.bootlin.com/linux/latest/source/arch/x86/kvm/vmx/vmx.c#L12

Reply:
Ah yes, I do not quite like the authorship in the comment header, it is quickly outdated...

Back to this file, the above UC San Diego copyright is added in the first commit in this series. I agree we can just leave one Google one in this file, so I will update the first commit with the Google copyright, if that sounds good.

Regarding the number of changes, there are more changes made to this file in follow-up patches, so I guess a copyright is OK given the total number of changes to this file?

Reply:
Let me finish going through the other patches :-). Hold off on changing the copyright for now. I'll also diff the two files with all patches applied after looking at the rest of the patches. I'll come back here with another comment after I go through the rest of them. Maybe my opinion will change shortly. Thanks for bearing with me :-).

Reply:
Haha no worry! I will be out for a few hours and come back to this PR tonight.

Reply:
It's very subjective. (See my other comment.) Personally, I wouldn't have added it in this case. Because `setup_tss()` was mostly developed at Google. And the five-level paging function is just adding a TODO. But I don't want to pressure you one way or the other. I'll leave it up to you. Please proceed however you see fit. Or feel free to add more discussion to make me feel more confident that the copyright is clearcut and appropriate :-). Digital communication is awkward. I'm honestly OK either way. But I myself wouldn't have added it.

Reply:
I thought about this more, and I think I should have more of an open mind. You're preparing the patches on your own time. And my thought process was very biased. (I really probably shouldn't even be commenting on it.) Please feel free to leave the UCSD copyright. I'm fine with it either way. I don't have a strong opinion either way anymore. Thanks!

Reply:
I think I will label all commits and copyrights with the Google email just to reduce the discrepancies. It's 100% OK to me as the Google email also represents me, so don't worry. Doing so may also prevent potential questions from the upstreaming community :-)